src/adapters/IDB.js

```diff
@@ -39,70 +116,70 @@ const cursorHandlers = {
       }
       const { key, value } = cursor;
       let i = 0;
-      while (key > sortedValues[i]) {
+      while (key > values[i]) {
         // The cursor has passed beyond this key. Check next.
         ++i;
-        if (i === sortedValues.length) {
+        if (i === values.length) {
           done(results); // There is no next. Stop searching.
           return;
         }
       }
-      if (key === sortedValues[i]) {
+      if (arrayEqual(key, values[i])) {
         results.push(value);
         cursor.continue();
       } else {
-        cursor.continue(sortedValues[i]);
+        cursor.continue(values[i]);
       }
     };
   },
 };
 
-/**
- * Extract from filters definition the first indexed field. Since indexes were
- * created on single-columns, extracting a single one makes sense.
- *
- * @param  {Object} filters The filters object.
- * @return {String|undefined}
- */
-function findIndexedField(filters) {
-  const filteredFields = Object.keys(filters);
-  const indexedFields = filteredFields.filter(field => {
-    return INDEXED_FIELDS.includes(field);
-  });
-  return indexedFields[0];
-}
-
 /**
  * Creates an IDB request and attach it the appropriate cursor event handler to
  * perform a list query.
  *
  * Multiple matching values are handled by passing an array.
  *
+ * @param  {String}           cid        The collection id (ie. `{bid}/{cid}`)
  * @param  {IDBStore}         store      The IDB store.
- * @param  {String|undefined} indexField The indexed field to query, if any.
- * @param  {Any}              value      The value to filter, if any.
- * @param  {Object}           filters    More filters.
+ * @param  {Object}           filters    Filter the records by field.
  * @param  {Function}         done       The operation completion handler.
  * @return {IDBRequest}
  */
-function createListRequest(store, indexField, value, filters, done) {
+function createListRequest(cid, store, filters, done) {
+  // Introspect filters and check if they leverage an indexed field.
+  const indexField = Object.keys(filters).filter(field => {
+    return INDEXED_FIELDS.includes(field);
+  })[0];
+
   if (!indexField) {
     // Get all records.
-    const request = store.openCursor();
+    const request = store.index("cid").openCursor(IDBKeyRange.only(cid));
     request.onsuccess = cursorHandlers.all(filters, done);
     return request;
   }
 
+  const value = filters[indexField];
```

Comment:
This seems dangerous. `indexField` could be `undefined` and who knows what should happen when you do `someMap[undefined]`? I'm pretty sure the output of `someMap[undefined]` is `undefined` but `value` is used in a bunch of other things later.

Reply:
we have `if (!indexField) {` above.